flang/lib/Lower/CallInterface.cpp

```diff
@@ -494,7 +496,7 @@ class Fortran::lower::CallInterfaceImpl {
           Fortran::common::visitors{
               [&](const auto &dummy) {
                 const auto &entity = getDataObjectEntity(std::get<1>(pair));
-                handleImplicitDummy(dummy, entity);
+                handleImplicitDummy(&std::get<0>(pair), dummy, entity);
```

Comment:
Can we give this `std::get<0>()` a name that tells us more about what this is doing?

Reply:
Named `argCharacteristics`. I named it using `argCharacteristics = std::get<0>()` because `for( auto [name1, name2] : llvm::zip(...))` cannot be used here since names defined in `[name1, name2]` cannot be  captured in lambdas (apparently, they are weird C++ things).